src/components/TodoElement.tsx

```diff
@@ -0,0 +1,169 @@
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodoContext } from '../Context/TodoContext';
+import { CurrentError } from '../types/CurrentError';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoElement: React.FC<Props> = ({ todo }) => {
+  const { completed, title, id } = todo;
+  const {
+    handleTodoDelete,
+    processingTodoIds,
+    handleTodoRename,
+    setError,
+    handleToggleChange,
+  } = useContext(TodoContext);
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(title);
+  const shouldDisplayLoader = id === 0 || processingTodoIds.includes(id);
+  const titleInputRef = useRef<HTMLInputElement | null>(null);
+
+  const handleTodoDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleTodoChange = async (
+    event: React.FormEvent<HTMLFormElement>,
+  ) => {
+    event.preventDefault();
+
+    try {
+      if (todoTitle) {
+        await handleTodoRename(todo, todoTitle);
+      } else {
+        await handleTodoDelete(id);
+      }
+
+      setIsEditing(false);
+    } catch (error) {
+      titleInputRef.current?.focus();
+
+      if (error === CurrentError.UpdateError) {
+        setError(CurrentError.UpdateError);
+        throw new Error();
+      }
+
+      if (error === CurrentError.DeleteError) {
+        setError(CurrentError.DeleteError);
+        throw new Error();
+      }
+
+      if (error === CurrentError.EmptyTitleError) {
+        setError(CurrentError.EmptyTitleError);
+        throw new Error();
+      }
```

Comment:
You don't need this as you set errors previously in you handlers, so you don't need to throw errors and handle them here one more time